platzky/plugin/plugin_loader.py

```
@@ -87,119 +171,87 @@ def _process_legacy_plugin(
+    except ValidationError as e:
+        raise PluginError(f"Invalid plugin configuration in database: {e}") from e
 
-    for plugin_data in plugins_data:
-        plugin_config = plugin_data["config"]
-        plugin_name = plugin_data["name"]
+    discovered: dict[str, type[PluginBase]] = {}
+    failed_entry_points: dict[str, Exception] = {}
+    for entry_point in importlib.metadata.entry_points(group=_ENTRY_POINT_GROUP):
+        try:
+            plugin_class = entry_point.load()
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)